src/shapes/object.class.js

```diff
@@ -979,6 +979,18 @@
       return opacity;
     },
 
+    /**
+     * Returns the object angle relative to canvas counting also the group property
+     * @returns {number}
+     */
+    getTotalAngle: function () {
+      var angle = this.angle;
+      if (this.group) {
+        angle += this.group.getTotalAngle();
+      }
+      return angle;
+    },
```

Comment:
i would use the qrDecompose to find the angle.
How this angle adding plays with skew and flip we don't know.